### PR TITLE
Draft: Write introduction, add implementation

### PR DESCRIPTION
**Pull request by dominikbell**
- Write an introduction based on Erics handwritten notes and the literature
- remove some trailing whitespaces
- (maybe) write about implementation

### CODE REVIEW

File: main.tex [hunk 1]
```diff
@@ -62,12 +62,12 @@
...
-	Après une introduction des équations gyro-cinétiques dans la Section \ref{sec:introduction}, nous décrirons cet ansatz de fractionnement en combinaison avec les deux schémas dans la Section \ref{sec:splitting_discretization}. Enfin, dans la section \ref{sec:num_exp}, des expériences numériques seront présentées, d'abord pour vérifier le schéma AKW, puis pour l'appliquer au modèle PyGyro complet. Dans la section \ref{sec:conclusion}, nous discutons des avantages de cette approche. 
+	Après une introduction des équations gyro-cinétiques dans la Section \ref{sec:gk-model}, nous décrirons cet ansatz de fractionnement en combinaison avec les deux schémas dans la Section \ref{sec:splitting_discretization}. Enfin, dans la section \ref{sec:num_exp}, des expériences numériques seront présentées, d'abord pour vérifier le schéma AKW, puis pour l'appliquer au modèle PyGyro complet. Dans la section \ref{sec:conclusion}, nous discutons des avantages de cette approche.
```
Comment (EmilyBourne) [inline main.tex:70]:
```suggestion
	Après une introduction des équations gyro-cinétiques dans la Section \ref{sec:gk-model}, nous décrirons cet ansatz de fractionnement en combinaison avec les deux schémas dans la Section \ref{sec:splitting_discretization}. Enfin, dans la Section \ref{sec:num_exp}, des expériences numériques seront présentées, d'abord pour vérifier le schéma AKW, puis pour l'appliquer au modèle PyGyro complet. Dans la Section \ref{sec:conclusion}, nous discutons des avantages de cette approche. 

```

File: main.tex [hunk 2]
```diff
@@ -62,12 +62,12 @@
 
 \begin{abstract}
 In practice, there are various ways to solve gyro-kinetic equations. Here, we will combine the Semi-Lagrangian (SL) method and the Arakawa (AKW) scheme combined with a time-integrator. Both methods are successfully used in practice when applied individually, in our case, we combine them by first decomposing the problem into a fast (parallel) and a slow (perpendicular) dynamical system. The SL approach will be used to solve the fast and the AKW scheme for the slow subsystem. So far in our reference code \cite{pygyro_code}, the entire model is solved only using the SL method. Our goal is to replace the discretization of the slow subsystem by the AKW scheme, in order to improve preservation of conserved quantities of the physical system at hand. \\
```
Comment (EmilyBourne) [inline main.tex:64]:
```suggestion
In practice, there are various ways to solve gyro-kinetic equations. Here, we will combine the Semi-Lagrangian (SL) method and the Arakawa (AKW) scheme combined with a time-integrator. Both methods are successfully used in practice when applied individually, in our case, we combine them by first decomposing the problem into a fast (parallel) and a slow (perpendicular) dynamical system. The SL approach will be used to solve the fast subsystem, and the AKW scheme for the slow subsystem. So far in our reference code \cite{pygyro_code}, the entire model is solved only using the SL method. Our goal is to replace the method used in the slow subsystem by the AKW scheme, in order to improve the conservation of the physical constants. \\

```

Not sure about the repetition between the last 2 sentences

**Comment by EmilyBourne** [comment]
I think you should talk about `backwards semi-Lagrangian (BSL)`, instead of `Semi-Lagrangian (SL)`

File: main.tex (hunk 2 above)
Comment (EmilyBourne) [inline main.tex:64]: > Here, we will combine the Semi-Lagrangian (SL) method and the Arakawa (AKW) scheme combined with a time-integrator.

Repetition of "combine". Do you mean:
> Here, we will combine the Semi-Lagrangian (SL) method and the Arakawa (AKW) scheme with a time-integrator.

or:

> Here, we will combine the Semi-Lagrangian (SL) method with a subcycling Arakawa (AKW) scheme.

Comment (EmilyBourne) [inline main.tex:64]: > Both methods are successfully used in practice when applied individually,

Not sure what is meant by this

Comment (EmilyBourne) [reply main.tex:64]: Or:
```suggestion
The SL approach and the AKW scheme will be used to solve respectively the fast and the slow subsystems. 
```

File: main.tex [hunk 3]
```diff
@@ -62,12 +62,12 @@
...
-After an introduction of the gyro-kinetic equations in Section \ref{sec:introduction}, we will describe this splitting ansatz in combination with the two schemes in Section \ref{sec:splitting_discretization}. Lastly in Section \ref{sec:num_exp}, numerical experiments will be presented, first as a verification of the AKW scheme, and after that the application to the full PyGyro model. Concluding with Section \ref{sec:conclusion}, we discuss the benefits of this approach. 
+After an introduction of the gyro-kinetic equations in Section \ref{sec:gk-model}, we will describe this splitting ansatz in combination with the two schemes in Section \ref{sec:splitting_discretization}. Lastly in Section \ref{sec:num_exp}, numerical experiments will be presented, first as a verification of the AKW scheme, and after that the application to the full PyGyro model. Concluding with Section \ref{sec:conclusion}, we discuss the benefits of this approach.
```
Comment (EmilyBourne) [inline main.tex:65]: This line feels like it should be at the end of the introduction not the abstract.

File: sections/conclusion.tex [hunk 4]
```diff
@@ -1,4 +1,14 @@
...
+
 \section{Conclusion}
 \label{sec:conclusion}
 
-Bonne travaille, je suis très satisfée.
\ No newline at end of file
+%TODO: or to keep this?
+%Bonne travaille, je suis très satisfée.
+
+In this work we demonstrated that the Arakawa scheme yields superior conservation properties of a numerical when implemented as the spatial discretization of the Poisson bracket in a gyro-kinetic model, compared to a Semi-Lagrangian scheme. Although the mass and $L^2$-norm of the distribution function, as well as the kinetic and potential energy are not preserved up to machine precision, their error occurring in the poloidal advection step is multiple orders of magnitude lower than when using a Semi-Lagrangian scheme.
```
Comment (EmilyBourne) [inline sections/conclusion.tex:9]: > superior conservation properties of a numerical

?

Comment (EmilyBourne) [inline sections/conclusion.tex:9]:
```suggestion
In this work we demonstrated that the Arakawa scheme yields superior conservation properties of a numerical when implemented as the spatial discretization of the Poisson bracket in a gyro-kinetic model, compared to a Semi-Lagrangian scheme. Although the mass and $L^2$-norm of the distribution function, as well as the kinetic and potential energy are not preserved up to machine precision, the error occurring in the poloidal advection step is multiple orders of magnitude lower than when using a Semi-Lagrangian scheme.
```

Comment (dominikbell) [reply sections/conclusion.tex:9]: The conservation of mass, L2-norm and energies is better, meaning the error that that occurs during the poloidal advection step is smaller.

File: sections/introduction.tex [hunk 5]
```diff
@@ -2,4 +2,12 @@
...
+
+This paper follows the latter method but replaces the Semi-Lagrangian scheme in the slow subsystem (which does not involve the magnetic field) by an Arakawa scheme. For this method we test different orders and boundary conditions. This spatial discretization will be combined with either an implicit Crank-Nicolson integrator or an explicit Runge-Kutta scheme of order 4. The implementation proceeds on top of the \texttt{PyGyro} code \cite{pygyro_code} which is a \textit{python} implementation of the \texttt{GYSELA} code.
```
Comment (EmilyBourne) [inline sections/introduction.tex:9]:
```suggestion
This paper follows the latter method but replaces the Semi-Lagrangian scheme in the slow subsystem (which does not involve the magnetic field) by an Arakawa scheme. For this method we test different orders and boundary conditions. This spatial discretization will be combined with either an implicit Crank-Nicolson integrator or an explicit Runge-Kutta scheme of order 4. The implementation proceeds on top of the \texttt{PyGyro} code \cite{pygyro_code} which is a \textit{Python} implementation of the \texttt{GYSELA} code.
```

Comment (EmilyBourne) [inline sections/introduction.tex:9]: PyGyro is **not** a python implementation of the GYSELA code. It is a python implementation of the code in Latu_2017. This code aims to replicate the test case used in the 2006 GYSELA paper, but GYSELA is **very** different now (16 years later). It has collision operators, an extra dimension, different geometry, etc.

File: sections/gyro-kinetic_model.tex [hunk 6]
```diff
@@ -2,7 +2,7 @@
 
 
 \section{The Gyro-kinetic Model}
-\label{sec:introduction}
+\label{sec:gk-model}
 
 In a Tokamak, particles have complicated dynamics in real space which consists of a slow motion along the magnetic field lines superimposed with a fast motion circling around the magnetic field lines. This fast motion is called gyration and can be averaged out to reduce the dimension of phase space in order to make the models computationally more feasible, while keeping most of the important physics. The resulting theory is called a gyro-kinetic model which will be discussed in the following.\\
```
Comment (EmilyBourne) [inline sections/gyro-kinetic_model.tex:7]: Not entirely true. I think there is slightly more involved in the gyro-kinetic model. This describes a guiding-centre model. There are some additional corrections to reach a gyro-kinetic model (maybe ask Martin).

File: sections/conclusion.tex (hunk 4 above)
Comment (EmilyBourne) [reply sections/conclusion.tex:9]: :+1: this is clearer. The current sentence is a fragment. I assume you forgot to finish writing it

File: sections/gyro-kinetic_model.tex [hunk 7]
```diff
@@ -83,28 +81,28 @@ \section{The Gyro-kinetic Model}
...
+%TODO add sources?
 \begin{equation}
 \partial_t f + \{\phi, f\} + v_\parallel \nabla_\parallel f - \nabla_\parallel \phi \partial_{v_\parallel}f = 0, \label{eq:GK_model}
 \end{equation}
-where $\nabla_\parallel = \bb \cdot \nabla$ and the bracket is transformed to polar coordinates, which reads, given the toroidal magnetic field $B_0$, 
+where $\nabla_\parallel = \bb \cdot \nabla$ and the bracket is transformed to polar coordinates, which reads, given the toroidal magnetic field $B_0$,
 \begin{equation}
 \{\phi, f \} = \frac{1}{rB_0}\partial_r \phi\partial_\theta f -\frac{1}{rB_0}\partial_\theta \phi\partial_r f. \label{eq:bracket_in_polar}
 \end{equation}
 Since the plasma is quasi-neutral, what means that locally there may be charged regions but it is neutral overall, this equation is complemented by solving the Poisson type quasi-neutrality (QN) equation for the self-consistent potential $\phi = \phi(t, r, \phi, z)$, i.e. solving for a given temperature profile $T_e$
```
Comment (EmilyBourne) [inline sections/gyro-kinetic_model.tex:100]:
```suggestion
Since the plasma is quasi-neutral, which means that locally there may be charged regions but it is neutral overall, this equation is complemented by solving the Poisson type quasi-neutrality (QN) equation for the self-consistent potential $\phi = \phi(t, r, \phi, z)$, i.e. solving for a given temperature profile $T_e$

```

File: sections/gyro-kinetic_model.tex [hunk 8]
```diff
@@ -83,28 +81,28 @@ \section{The Gyro-kinetic Model}
...
+where the given (radial symmetric) equilibrium function $f_\text{eq}$ is a Gaussian and the radial profile $n_0$ which is the integral over $\vp$ of the equilibrium function. The initial distribution function $f(t=0, r, \theta, z, v_\parallel)$ is defined to be the equilibrium with a perturbation in some modes, which will result in an observable turbulence behaviour of the system after a certain amount of time. For more details and the exact definitions and constants, we refer to \cite{Latu_2017}.
```
Comment (EmilyBourne) [inline sections/gyro-kinetic_model.tex:104]:
```suggestion
where the given (radial symmetric) equilibrium function $f_\text{eq}$ is a Gaussian and the radial profile $n_0$ is the integral over $\vp$ of the equilibrium function. The initial distribution function $f(t=0, r, \theta, z, v_\parallel)$ is defined to be the equilibrium with a perturbation in some modes, which will result in an observable turbulence behaviour of the system after a certain amount of time. For more details and the exact definitions and constants, we refer to \cite{Latu_2017}.

```

Comment (EmilyBourne) [inline sections/gyro-kinetic_model.tex:104]: There are multiple test cases in \cite{Latu_2017} (section 4 and 5) so I'm not sure that this explanation is clear enough to allow the reader to reproduce the test case. I think it would be worth writing the initial conditions here

File: sections/gyro-kinetic_model.tex [hunk 9]
```diff
@@ -83,28 +81,28 @@ \section{The Gyro-kinetic Model}
...
+To finalize this system of equations, we briefly want to discuss the boundary conditions of the screw-pinch model. The distribution function $f$ is periodic along $\theta$, $z$, while having homogeneous Dirichlet boundaries in $\vp$-direction. In the radial direction $r$, we assume the values are given by an outside equilibrium function. For the potential $\phi$, we assume periodic boundary conditions along $\theta$ and $z$. In $r$-direction, we decompose $\phi$ into Fourier modes at $r_\text{min}$, taking homogeneous Neumann boundary conditions for the zeroth mode and homogeneous Dirichlet boundary conditions for the others. At $r_\text{max}$, we just take plain homogeneous Dirichlet boundary conditions. \\
```
Comment (EmilyBourne) [inline sections/gyro-kinetic_model.tex:105]:
```suggestion
To complete this system of equations, we briefly want to discuss the boundary conditions of the screw-pinch model. The distribution function $f$ is periodic along $\theta$, $z$, while having homogeneous Dirichlet boundaries in $\vp$-direction. In the radial direction $r$, we assume the values are given by an outside equilibrium function. For the potential $\phi$, we assume periodic boundary conditions along $\theta$ and $z$. In $r$-direction, we decompose $\phi$ into Fourier modes at $r_\text{min}$, taking homogeneous Neumann boundary conditions for the zeroth mode and homogeneous Dirichlet boundary conditions for the others. At $r_\text{max}$, we just take plain homogeneous Dirichlet boundary conditions. \\

```

File: sections/gyro-kinetic_model.tex [hunk 10]
```diff
@@ -113,11 +111,11 @@ \section{The Gyro-kinetic Model}
 \begin{equation}
 \dt C(f) + \bracket{C(f)}{H} =0,
 \end{equation}
-where the Casimir invariant $\int C( f ) d^5R$ is conserved. Therefore, in addition to the particle number or the $L^1$-norm $\int f d^5R$, the system has an infinite number of conserved quantities such as the $L^2$-norm $\int f^2 d^5R$, and the kinetic entropy $S=\int f \ln(f) d^5R$.\\
+where the Casimir invariant $\int C( f ) d^5R$ is conserved. Therefore, in addition to the particle number or the $L^1$-norm $\int f d^5R$, the system has an infinite number of conserved quantities such as the $L^2$-norm $\int f^2 d^5R$, and the kinetic entropy $S=\int f \ln(f) d^5R$. \\
 Thirdly, the gyro-kinetic equations conserve the total energy $H = E_k + E_f$:
```
Comment (EmilyBourne) [inline sections/gyro-kinetic_model.tex:115]: What is first and second?

**Comment by EmilyBourne** [comment]
[where $E_k$ is the kinetic energy and $E_f$ is the field energy. For more details we refer the to \cite{idomura2008conservative}.](https://github.com/dominikbell/cemracs_paper/blob/f835dba452737b579b68b1c1246a5a25406dd303/sections/gyro-kinetic_model.tex#L122)

missing word:

> where $E_k$ is the kinetic energy and $E_f$ is the field energy. For more details we refer the reader to \cite{idomura2008conservative}.

File: sections/operator_splitting_and_discretization.tex [hunk 11]
```diff
@@ -3,7 +3,7 @@
...
+Operator splitting has proven to be an effective method for the time-integration of ordinary differential equations (ODEs), whose vector-field can be written as a sum of simpler terms. This is of special interest for geometric integration, that is, if the underlying problem has geometric properties that should be conserved by the integrator. Furthermore, the computational cost goes down, while we introduce an additional approximation error. A broad overview of these methods is given in \cite{mclachlan_quispel_2002}, also containing the second-order Strang and first-order Lie splitting, that we will use in the following. \\
```
Comment (EmilyBourne) [inline sections/operator_splitting_and_discretization.tex:6]: Not sure about the use of "Furthermore" here. "Furthermore" implies that the conclusions are similar, but this seems to be mixing positive and negative comments:
+ effective method
+ special interest
+ computational cost goes down
- additional error

File: sections/operator_splitting_and_discretization.tex [hunk 12]
```diff
@@ -12,32 +12,32 @@ \section{Discretization via Operator Splitting}
...
 \end{equation}
-while the second-order Strang splitting is given by 
+while the second-order Strang splitting is given by
 \begin{equation}
 	f^{n+1} = A\left(\frac{\Delta \tau}{2}\right) B\left(\frac{\Delta \tau}{2}\right) C\left(\Delta \tau\right) B\left(\frac{\Delta \tau}{2}\right) A\left(\frac{\Delta \tau}{2}\right) f^n. \label{eq:Strang}
 \end{equation}
-So far, we assumed to have a good approximation of the potential $\phi$ at hand. In practice, this is obtained by solving the QN equation i.e. \eqref{eq:qn} by a spectral Finite Element method (FEM) combined with some Finite Difference (FD) approximations. Since this is not the main part of our work, we refer to the sources above for more details. 
+So far, we assumed to have a good approximation of the potential $\phi$ at hand. In practice, this is obtained by solving the QN equation i.e. \eqref{eq:qn} by a spectral Finite Element method (FEM) combined with some Finite Difference (FD) approximations. Since this is not the main part of our work, we refer to the sources above for more details.
```
Comment (EmilyBourne) [inline sections/operator_splitting_and_discretization.tex:29]:
```suggestion
So far, we assumed that we have a good approximation of the potential $\phi$ to hand. In practice, this is obtained by solving the QN equation i.e. \eqref{eq:qn} by a spectral Finite Element method (FEM) combined with some Finite Difference (FD) approximations. Since this is not the main part of our work, we refer to the sources above for more details.

```

Comment (EmilyBourne) [inline sections/operator_splitting_and_discretization.tex:29]: > \eqref{eq:qn} by a spectral Finite Element method (FEM) combined with some Finite Difference (FD) approximations

I don't know what you are referring to here. PyGyro uses a Fourier transform in the poloidal direction, and FEM in the radial direction. It uses the quasi-toroial approximation to reduce the problem to 1D (ie. no z/phi component). As far as I am aware there are no finite-differences here

File: sections/operator_splitting_and_discretization.tex [hunk 13]
```diff
@@ -12,32 +12,32 @@ \section{Discretization via Operator Splitting}
...
+This approach has shown to be quite successful, which inter alia is due to the unconditional stability of the SL scheme. Nonetheless, the method lacks structure preserving properties, like conservation of energy, mass and $L^2$-norm. This gives rise to the main idea of this project, which exchanges the easily obtainable stability properties of the SL method for the structure preservation of an AKW FD scheme, at least for the slow-time subsystem, where a time-step restriction is supposed to be not too computationally expensive. In other words, we solve the poloidal advection equation i.e. \eqref{eq:adv_poloidal} by applying the AKW scheme combined with a suitable time-integration. Since this advection equation essentially consists of a Poisson bracket, which is rich in geometric structure - exactly what the AKW scheme was designed for - we hope to improve the overall conservation properties of the full simulation.
```
Comment (EmilyBourne) [inline sections/operator_splitting_and_discretization.tex:40]: "inter alia"?

Comment (EmilyBourne) [inline sections/operator_splitting_and_discretization.tex:40]:
```suggestion
This approach has shown to be quite successful, which inter alia is due to the unconditional stability of the SL scheme. Nonetheless, the method lacks structure preserving properties, like conservation of energy, mass and $L^2$-norm. This gives rise to the main idea of this project, which exchanges the easily obtainable stability properties of the SL method for the structure preservation of an AKW FD scheme, at least for the slow-time subsystem, where a time-step restriction is supposed to not be too computationally expensive. In other words, we solve the poloidal advection equation i.e. \eqref{eq:adv_poloidal} by applying the AKW scheme combined with a suitable time-integration. Since this advection equation essentially consists of a Poisson bracket, which is rich in geometric structure - exactly what the AKW scheme was designed for - we hope to improve the overall conservation properties of the full simulation.

```

Comment (dominikbell) [reply sections/operator_splitting_and_discretization.tex:40]: "Among other things" in Latin `git blame` @FrederikSchnack

File: sections/operator_splitting_and_discretization.tex [hunk 14]
```diff
@@ -137,8 +140,8 @@ \subsubsection{Construction of the Discrete Bracket}
...
+%TODO: Do we want a picture here?
+Denoting this approximation by $J_1=\frac{1}{3}(J^{++}+J^{+\times}+J^{\times+})$, which is the only such second order approximation of the analytical Poisson bracket that conserves mass, $L^2$-norm and energy as is shown in \cite{Arakawa_1966}. The $+$ and $\times$ notation comes from the patterns done by the FD stencils on the grid, as is visualized in Figure \ref{fig:stencils}. So for example in $J^{+x}$, we choose the $+$-pattern, which corresponds to two central FD schemes, for the discretization of $\partial_x f_{i,j}$ and $\partial_y f_{i,j}$ and the same with the $\times$-pattern for $g_{i,j}$, then multiply the two together getting a discretization of the bracket \eqref{eq:poisson_bracket}.
```
Comment (EmilyBourne) [inline sections/operator_splitting_and_discretization.tex:144]:
```suggestion
Denoting this approximation by $J_1=\frac{1}{3}(J^{++}+J^{+\times}+J^{\times+})$, which is the only such second order approximation of the analytical Poisson bracket that conserves mass, $L^2$-norm and energy as is shown in \cite{Arakawa_1966}. The $+$ and $\times$ notation comes from the patterns done by the FD stencils on the grid, as is visualized in Figure \ref{fig:stencils}. So for example in $J^{+\times}$, we choose the $+$-pattern, which corresponds to two central FD schemes, for the discretization of $\partial_x f_{i,j}$ and $\partial_y f_{i,j}$ and the same with the $\times$-pattern for $g_{i,j}$, then multiply the two together getting a discretization of the bracket \eqref{eq:poisson_bracket}.

```

File: sections/operator_splitting_and_discretization.tex (hunk 12 above)
Comment (FrederikSchnack) [reply sections/operator_splitting_and_discretization.tex]: I think the confusion came from the comments right before 4.3 in the Latu_2017 paper and our mistake to check if PyGyro does  it differently. :sweat_smile:  
"For each mode m, this ordinary differential equation is collocated at the grid points r = ri ,
and the derivatives are approximated by 2nd-order central finite differences."

**Review by FrederikSchnack — APPROVED** [review]
lgtm, albeit some todos
